src/engine/strat_engine/liminal/setup.rs

```diff
@@ -281,7 +297,7 @@ pub fn get_blockdevs(
     fn check_and_sort_devs(
         mut devs: Vec<StratBlockDev>,
         dev_map: &HashMap<DevUuid, (usize, &BaseBlockDevSave)>,
-    ) -> StratisResult<Vec<StratBlockDev>> {
+    ) -> BDARecordResult<Vec<StratBlockDev>> {
```

Comment:
Suggest that you not change ```check_and_sort_devs``` at all for this PR. Instead, if ```check_and_sort_devs``` returns an error, then you know that you need to pull the BDAs out of the blockdevs + cachedevs combined and you can do that easily at line 345 and line 359.

I think that way is clearer and just a little more robust, too.

Reply:
That doesn't work. `check_and_sort_devs` consumes the vector so we either need to return `Vec<StratBlockdev>` or `HashMap<DevUuid, BDA>` in the `Err(_)` variant. The best we can do is pass a mutable reference to `check_and_sort_devs` which still requires a change.

Reply:
Whoops! Yes, that's unavoidable. I no longer have a concern about this code so you should feel free to carry on.

Reply:
Ok!